src/main/webapp/app/course/competencies/competency-accordion/competency-accordion.component.ts

```diff
@@ -89,23 +90,36 @@ export class CompetencyAccordionComponent implements OnChanges {
             this.nextExercises = [];
         }
 
-        const submittedExercises = Object.keys(this.metrics.exerciseMetrics?.latestSubmission ?? {}).map(Number);
-        const competencyExercises = this.metrics.competencyMetrics?.exercises?.[this.competency.id] ?? [];
-        const nextExerciseInformations = competencyExercises
-            .filter((exerciseId) => !submittedExercises.includes(exerciseId))
-            .flatMap((exerciseId) => this.metrics.exerciseMetrics?.exerciseInformation?.[exerciseId] ?? [])
-            .filter((exercise) => exercise.startDate.isBefore(dayjs()) && (!exercise.dueDate || exercise.dueDate.isAfter(dayjs())))
-            .sort((a, b) => (a.dueDate ?? a.startDate).diff(b.dueDate))
-            .slice(0, 5);
-
-        // Workaround to convert ExerciseInformation to Exercise
-        this.nextExercises = nextExerciseInformations.map(
-            (exercise) =>
-                ({
-                    ...exercise,
-                    studentAssignedTeamIdComputed: exercise.studentAssignedTeamId !== undefined,
-                }) as unknown as Exercise,
+        const courseExercises = this.course?.exercises ?? [];
+        const exerciseIdToExercise = Object.fromEntries(courseExercises.map((exercise) => [exercise.id, exercise] as [number, Exercise]));
+        const activeCompetencyExercises = (this.metrics.competencyMetrics?.exercises?.[this.competency.id] ?? [])
+            .flatMap((exerciseId) => [exerciseIdToExercise[exerciseId]] ?? [])
+            .filter((exercise) => exercise.releaseDate?.isBefore(dayjs()))
+            .filter((exercise) => exercise.dueDate?.isAfter(dayjs()) || isStartPracticeAvailable(exercise));
+
+        const exerciseIdToMaxScore = Object.fromEntries(
+            activeCompetencyExercises.map((exercise) => {
+                const score =
+                    exercise.studentParticipations?.flatMap((participation) => participation.results ?? []).reduce((max, result) => Math.max(max, result.score ?? 0), -1) ?? 0;
+                return [exercise.id, score] as [number, number];
+            }),
         );
+
+        const completionThreshold = 80;
```

Comment:
```suggestion
        const completionThreshold = MIN_SCORE_GREEN;
```
We have a constant at which score the results should be considered successful and shown in green. I think this also applies here